packages/scenes/src/components/VizPanel/VizPanelRenderer.tsx

```diff
@@ -66,6 +67,10 @@ export function VizPanelRenderer({ model }: SceneComponentProps<VizPanel>) {
   if (model.state.$timeRange) {
     titleItems.push(<model.state.$timeRange.Component model={model.state.$timeRange} />);
   }
+  let panelMenu;
+  if (menu) {
+    panelMenu = <menu.Component model={menu} />;
```

Comment:
```suggestion
    panelMenu = () => <menu.Component model={menu} />;
```

I think this should be lazy, so it's only rendered when it's opened

Reply:
@dprokop this is the important bit I think ^ 
that way you can use the VizPanelMenu's activation handler to do things when the menu opens

Reply:
@dprokop just tested this, it's not really needed. This component is only rendered when menu is shown anyway, so this comment was wrong

Reply:
Precisely, that was done on purpose.